packages/api/endpoints/elasticsearch.js

```diff
@@ -0,0 +1,186 @@
+  // }
+  // catch (err) {
+  //   // Handle repository missing exceptions
+  //   if (!err.message.includes('[repository_missing_exception]')) {
+  //     throw err;
+  //   }
+
+  // TO DO: when permission boundaries are updated
+  // repository = await esClient.snapshot.createRepository({
+  //   repository: snapshotRepoName,
+  //   verify: false,
+  //   body: {
+  //     type: 's3',
+  //     settings: {
+  //       bucket: 'lf-internal',
+  //       region: 'us-east-1',
+  //       role_arn: process.env.ROLE_ARN
+  //     }
+  //   }
+  // });
+  // }
+}
+
+async function reindex(req, res) {
+  let sourceIndex = req.body.sourceIndex;
+  let destIndex = req.body.destIndex;
+  const aliasName = req.body.aliasName || defaultIndexAlias;
+
+  const esClient = await Search.es();
+
+  const alias = await esClient.indices.getAlias({
+    name: aliasName
+  });
+
+  // alias keys = index name
+  const indices = Object.keys(alias);
+
+  if (!sourceIndex) {
+    if (indices.length > 1) {
+      // We don't know which index to use as the source, throw error
+      return res.boom.badRequest(`Multiple indices found for alias ${aliasName}. Specify source index as one of [${indices.sort().join(', ')}].`);
+    }
+
+    sourceIndex = indices[0];
+  } else {
+    const sourceExists = await esClient.indices.exists({ index: sourceIndex });
+
+    if (!sourceExists) {
+      return res.boom.badRequest(`Source index ${sourceIndex} does not exist.`);
+    }
+
+    if (indices.includes(sourceIndex) === false) {
+      return res.boom.badRequest(`Source index ${sourceIndex} is not aliased with alias ${aliasName}.`);
+    }
+  }
+
+  if (!destIndex) {
+    const date = new Date();
+    destIndex = `cumulus-${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+  }
+
+  const destExists = await esClient.indices.exists({ index: destIndex });
+
+  if (destExists) {
+    return res.boom.badRequest(`Destination index ${destIndex} exists. Please specify an index name that does not exist.`);
+  }
+
+  // create destination index
+  await esClient.indices.create({
+    index: destIndex,
+    body: { mappings }
+  });
+
+  log.info(`Created destination index ${destIndex}.`);
+
+  // reindex
+  esClient.reindex({
+    body: {
+      source: { index: sourceIndex },
+      dest: { index: destIndex }
+    }
+  });
+
+  const message = `Reindexing to ${destIndex} from ${sourceIndex}. Check the reindex-status endpoint for status.`;
+
+  return res.status(200).send({ message });
+}
+
+async function reindexStatus(req, res) {
+  const esClient = await Search.es();
+
+  const reindexTaskStatus = await esClient.tasks.list({ actions: ['*reindex'] });
+
+  await esClient.indices.refresh();
+
+  const indexStatus = await esClient.indices.stats({
+    metric: 'docs'
+  });
+
+  const status = {
+    reindexStatus: reindexTaskStatus,
+    indexStatus
+  };
+
+  return res.send(status);
+}
+
+async function completeReindex(req, res) {
```

Comment:
I wonder if `changeIndex` is a clearer name for what this is doing. Because it's copying the existing source index to a new destination index and pointing the alias at the new destination index, correct?

Reply:
Yep, changed it to `changeIndex`. I think that's better because it could apply somehow outside the context of reindexing. Also you could use it to change the index back if your reindex went badly. Will update the doc PR as well.